src/fu-engine.c

```diff
@@ -4310,6 +4387,8 @@ fu_engine_add_device (FuEngine *self, FuDevice *device)
 
 	/* sometimes inherit flags from recent history */
 	fu_engine_device_inherit_history (self, device);
+
+	fu_engine_update_motd (self, NULL);
```

Comment:
I think it would be better to call something like `fu_engine_update_list_changed()` which in turn calls `fu_engine_update_motd()` as I'm sure we'll be using this for other things in the future.

Reply:
And we can also handle the motd failure in `fu_engine_update_list_changed` rather than in all the callers....

Reply:
It also feels like the kind of thing we should have disable-able in daemon.conf too...

Reply:
done